lib/spack/spack/repo.py

```diff
@@ -1646,6 +1654,73 @@ def fetched(self) -> bool:
             return self._fetched()
         return False
 
+    def _init_git_repo(self, git: spack.util.executable.Executable, remote: str):
+        git("init", self.destination, output=str)
+        git("-C", self.destination, "remote", "add", "origin", self.repository)
```

Comment:
Elsewhere we avoid `-C` because older git doesn't support that.

Reply:
Done.